src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,128 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private int size;
+    private T[] listArray;
+
+    public ArrayList() {
+        size = 0;
+        listArray = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+    }
+
     @Override
     public void add(T value) {
+        if (size >= listArray.length) {
+            listArray = extendArray();
+        }
 
+        listArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size + 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
+
+        if (size + 1 >= listArray.length) {
+            listArray = extendArray();
+        }
 
+        size++;
```

Comment:
The 'size++' should be moved after the newArray is filled, because if an exception occurs while filling the newArray, the size variable will be incorrect.